unit_tests/fvm/AnalyticalFunction.cpp

```diff
@@ -0,0 +1,120 @@
+#include "AnalyticalFunction.H"
+
+namespace amr_wind_tests {
+
+AnalyticalFunction::AnalyticalFunction(int degree, int ncomponents)
+    : m_degree(degree), m_ncomponents(ncomponents)
+{
+    // fixme do we want random variables assigned here?
+    // fixme do we want different coeffs for each component?
+    m_coeffs.assign(std::pow(degree + 1, AMREX_SPACEDIM), 2.0);
+}
+
+AnalyticalFunction::~AnalyticalFunction() = default;
+
+amrex::Real AnalyticalFunction::phi_eval(
+    amrex::Real x, amrex::Real y, amrex::Real z, int component) const
+{
+    amrex::Vector<amrex::Real> phi;
+    phi.assign(m_ncomponents, 0.0);
```

Comment:
`phi_eval` needs to be a device function, based on how it is used in the test. We cannot do this declaration if this is a device function.